src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -101,6 +101,16 @@ if (__DEV__) {
       ctor && (ctor.displayName || ctor.name) || 'ReactClass'
     );
   };
+
+  var warnAboutUpdateInRender = function(instance : ReactClass<any>) {
+    warning(
+      ReactCurrentOwner.current == null,
```

Comment:
This was always a bad pattern before because it assumed something about the "owner" pattern being a complete way to track this. It is also unnecessarily a deep dependency into an isomorphic library that can't be deduped. I believe that it is also not completely safe because `ReactCurrentOwner.current` doesn't get immediately reset to `null` like it does in Stack.

It would be better to track a DEV only specific flag that is designed for exactly this use case.